src/main/java/com/zendesk/maxwell/producer/MaxwellRedisProducer.java

```diff
@@ -31,20 +35,27 @@ public MaxwellRedisProducer(MaxwellContext context) {
 		this.interpolateChannel = channel.contains("%{");
 		this.redisType = context.getConfig().redisType;
 
-		initRedisConnection();
-
-		if (context.getConfig().redisAuth != null) {
-			jedis.auth(context.getConfig().redisAuth);
-		}
-
-		if (context.getConfig().redisDatabase > 0) {
-			jedis.select(context.getConfig().redisDatabase);
+		if (jedisPool == null) {
+			jedisPool = new JedisPool(
+				createdRedisPoolConfig(), 
+				context.getConfig().redisHost, 
+				context.getConfig().redisPort,
+				Protocol.DEFAULT_TIMEOUT,
+				context.getConfig().redisAuth, //even if not present jedispool will handle a null value 
+				context.getConfig().redisDatabase); //even if not present jedispool will handle a null value 
 		}
 	}
 
-	private void initRedisConnection() {
-		jedis = new Jedis(context.getConfig().redisHost, context.getConfig().redisPort);
-		jedis.connect();
+	private JedisPoolConfig createdRedisPoolConfig() {
```

Comment:
nitpick, maybe just name this `createRedisPoolConfig` or event `redisPoolConfig()`

Reply:
sorry for the typo